platform/consul/controller.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consul
+
+import (
+	"github.com/golang/glog"
+	"github.com/hashicorp/consul/api"
+	"istio.io/pilot/model"
+)
+
+type Controller struct {
```

Comment:
Since the plan is to have an Istio control plane per cluster (see https://groups.google.com/forum/#!topic/istio-dev/IZyAP17LVok), I believe it would make sense for the Consul registry adapter to bind to a particular data center.

Consul's definition of a data center:
> Datacenter - While the definition of a datacenter seems obvious, there are subtle details that must be considered. For example, in EC2, are multiple availability zones considered to comprise a single datacenter? *We define a datacenter to be a networking environment that is private, low latency, and high bandwidth.* This excludes communication that would traverse the public internet, but for our purposes multiple availability zones within a single EC2 region would be considered part of a single datacenter.
- https://www.consul.io/docs/internals/architecture.html#glossary